dhash/module.py

```diff
@@ -383,9 +438,66 @@ async def _save_hashes(self, message: nextcord.Message):
             )
             yield h
 
+    async def _url_hashes(self, message: nextcord.Message):
+        for url in re.findall(URL_REGEX, message.content):
+            if self.allowed_urls and not re.search(self.allowed_urls, url):
+                continue
+            try:
+                req = urllib.request.Request(url, headers=HEADERS)
+                file = urllib.request.urlopen(req, timeout=20)
+            except (URLError, HTTPError, ValueError) as ex:
+                await guild_log.debug(
+                    message.author,
+                    message.channel,
+                    f"DHash could not check url {url}.",
+                    exception=ex,
+                )
+                continue
+
+            size = file.headers.get("content-length")
+            if size:
+                size = int(size)
+
+            if not size or size > MAX_ATTACHMENT_SIZE * 1024:
+                continue
```

Comment:
You can't just call `continue`, you have `file` handle opened.

Reply:
Fixed.